main.cpp

```diff
@@ -18,13 +18,22 @@ using namespace clang::ast_matchers;
 using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
+private:
+    Rewriter& rewriter;
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
+    CastCallBack(Rewriter& rewriter) : rewriter(rewriter) {};
 
     void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
+        auto Expr = Result.Nodes.getNodeAs<CStyleCastExpr>("cast");
+        if (Expr) {
+            auto lborder = Expr->getLParenLoc();
+            auto rborder = Expr->getRParenLoc();
+            auto end = Expr->getEndLoc();
+
+            rewriter.ReplaceText(lborder, 1, "static_cast<");
+            rewriter.ReplaceText(rborder, 1, ">(");
+            rewriter.InsertTextAfterToken(end, ")");

```

Comment:
Please describe how to change the program code to get the following result:
```cpp
int main() {
    float fooooo;
    int i = static_cast<int>          (fooooo)      ;
    return 0;
}
```
 for the origin `test.cpp`:
```cpp
int main() {
    float fooooo;
    int i = (int)          fooooo      ;
    return 0;

}
```

Reply:
You need to change next code:
`rewriter.ReplaceText(rborder, 1, ">(");`
Like this:
`rewriter.ReplaceText(rborder, 1, ">");`
After this you need to run command:
`rewriter.InsertText(Expr->getSubExprAsWritten()->getBeginLoc(), "(");`
This code find first token of subexpression and place the open bracket.